modules/statistics/php/NDB_Form_statistics.class.inc

```diff
@@ -1014,27 +1050,75 @@ class NDB_Form_Statistics extends NDB_Form
             $projects[$key] = $value;
         }
 
-        if (isset($_REQUEST['ReliabilityProject'])
-            && $_REQUEST['ReliabilityProject'] != ''
-        ) {
-            $currentProject = $_REQUEST['ReliabilityProject'];
+        if ($_REQUEST['ReliabilityProject'] ?? '') {
+            $currentProject = htmlspecialchars($_REQUEST['ReliabilityProject']);
             $this->tpl_data['CurrentProject']
                 = array(
                    'ID'   => $currentProject,
                    'Name' => $projects[$currentProject],
                   );
             $Param_Project       ='AND (c.ProjectID IS NULL OR c.ProjectID=:pid) ';
-            $this->params['pid'] =$_REQUEST['ReliabilityProject'];
+            $this->params['pid'] = htmlspecialchars($_REQUEST['ReliabilityProject']);
         } else {
             $Param_Project ='';
         }
 
-        //SITES
-        if (isset($_REQUEST['ReliabilitySite'])
-            && $_REQUEST['ReliabilitySite'] != ''
+        //SUBPROJECTS
+        $subprojects[null] = 'All Valid Sub-Projects';
+
+        foreach (Utility::getSubprojectsForProject($currentProject)
+            as $key=>$value) {
+            $subprojects[$key] = $value;
+        }
+
+        foreach ($subprojects as $key=>$val) {
+            $subprojList .= $key.",";
```

Comment:
As far as I can tell, this is appending to a variable that doesn't exist at this point. I believe this will generate some PHP warnings.

I also don't see anywhere that this variable is used in the function, so this seems to be dead code that can just be removed.